apps/full-site-editing/full-site-editing-plugin/donations/blocks/donations/tabs.js

```diff
@@ -0,0 +1,147 @@
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import StripeNudge from './stripe-nudge';
+import intervalClassNames from '../donation/interval-class-names';
+
+const Tabs = ( { attributes, clientId, products, setAttributes, siteSlug, stripeConnectUrl } ) => {
+	const { showMonthly, showAnnually, showCustom } = attributes;
+	const [ activeTab, setActiveTab ] = useState( 'one-time' );
+
+	const isTabActive = ( tab ) => activeTab === tab;
+
+	const innerBlocks = useSelect(
+		( select ) => select( 'core/block-editor' ).getBlocks( clientId ),
+		[ clientId ]
+	);
+	const { replaceInnerBlocks } = useDispatch( 'core/block-editor' );
+
+	// Updates the inner blocks when settings are toggled.
+	useEffect( () => {
+		const newInnerBlocks = [];
+		const intervals = [
+			'one-time',
+			...( showMonthly ? [ '1 month' ] : [] ),
+			...( showAnnually ? [ '1 year' ] : [] ),
+		];
+		intervals.forEach( ( interval ) => {
+			const donationBlock = innerBlocks.find( ( block ) => block.attributes.interval === interval );
+			newInnerBlocks.push(
+				donationBlock ??
+					createBlock( 'a8c/donation', {
+						planId: products[ interval ],
+						interval,
+					} )
+			);
+		} );
+
+		if ( newInnerBlocks.length !== innerBlocks.length ) {
+			replaceInnerBlocks( clientId, newInnerBlocks, false );
+		}
+	}, [ innerBlocks, showMonthly, showAnnually, showCustom ] );
+
+	// Activates the one-time tab if the interval of the current active tab is disabled.
+	useEffect( () => {
+		if ( ! showMonthly && isTabActive( '1 month' ) ) {
+			setActiveTab( 'one-time' );
+		}
+
+		if ( ! showAnnually && isTabActive( '1 year' ) ) {
+			setActiveTab( 'one-time' );
+		}
+	}, [ showMonthly, showAnnually ] );
+
+	// Toggles the visibility of the inner blocks based on the current tab.
+	useEffect( () => {
+		const activeBlock = document.querySelector( '.wp-block-a8c-donation.is-active' );
+		if ( activeBlock ) {
+			activeBlock.classList.remove( 'is-active' );
+		}
+		const newActiveBlock = document.querySelector(
+			`.wp-block-a8c-donation.${ intervalClassNames[ activeTab ] }`
+		);
+		if ( newActiveBlock ) {
+			newActiveBlock.classList.add( 'is-active' );
+		}
+	}, [ activeTab, innerBlocks ] );
+
+	return (
+		<Block.div>
+			{ stripeConnectUrl && <StripeNudge stripeConnectUrl={ stripeConnectUrl } /> }
+			<div className="donations__container">
+				{ ( showMonthly || showAnnually ) && (
+					<div className="donations__tabs">
+						<Button
+							className={ classNames( { 'is-active': isTabActive( 'one-time' ) } ) }
+							onClick={ () => setActiveTab( 'one-time' ) }
+						>
+							{ __( 'One-Time', 'full-site-editing' ) }
+						</Button>
+						{ showMonthly && (
+							<Button
+								className={ classNames( { 'is-active': isTabActive( '1 month' ) } ) }
+								onClick={ () => setActiveTab( '1 month' ) }
+							>
+								{ __( 'Monthly', 'full-site-editing' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **69** times:
`translate( 'Monthly' )` *ES Score: 13*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/donation/customize-messages)